common/lib/dependabot/git_metadata_fetcher.rb

```diff
@@ -72,8 +72,14 @@ def fetch_upload_pack_for(uri)
       response = fetch_raw_upload_pack_for(uri)
       return response.body if response.status == 200
 
-      response_with_git = fetch_raw_upload_pack_with_git_for(uri)
-      return response_with_git.body if response_with_git.status == 200
+      # if git is installed, try fetching with it in case something we need is configured in .gitconfig (like creds)
+      env = { "PATH" => ENV.fetch("PATH", nil) }
+      command = SharedHelpers.escape_command("git --version")
+      stdout, stderr, process = Open3.capture3(env, command)
+      if process.success?
+        response = fetch_raw_upload_pack_with_git_for(uri)
+        return response.body if response.status == 200
```

Comment:
this may be better left as `response_with_git` so that the response doesn't hit the later error handling, I wasn't sure...